test/FunctionalTests/Balancer/ConnectionTests.cs

```diff
@@ -180,6 +180,55 @@ Task<HelloReply> UnaryMethod(HelloRequest request, ServerCallContext context)
         AssertHasLog(LogLevel.Trace, "ConnectingOnCreateStream", "Subchannel id '1' doesn't have a connected socket available. Connecting new stream socket for 127.0.0.1:50051.");
     }
 
+    [Test]
+    public async Task Active_UnaryCall_ServerCloseOnKeepAlive_SocketRecreatedOnRequest()
+    {
+        // Ignore errors
+        SetExpectedErrorsFilter(writeContext =>
+        {
+            return true;
+        });
+
+        Task<HelloReply> UnaryMethod(HelloRequest request, ServerCallContext context)
+        {
+            return Task.FromResult(new HelloReply { Message = request.Name });
+        }
+
+        // In this test the client connects to the server, and the server then closes it after keep-alive is triggered.
```

Comment:
Would the behavior be different if the server aborted the connection, rather than timing out?  AFAIK, the kestrel code paths are the same, so I'm guessing not?

Reply:
Depends on the kind of abort. A HTTP/2 abort would also send a GOAWAY. A transport-level aport would just close the socket. There are tests that already cover a transport-level abort.